### DISCUSSION TITLE
Illuminate\Database\Query\JoinClause::onRaw

### DISCUSSION THREAD

**Discussion by wimski** [laravel/framework#46927] [Ideas]
At the moment the `JoinClause` class only supports `on` (and `orOn`) which always uses `whereColumn` under the hood. This means you cannot use any raw statements as your 'on' clauses. Would it be possible to add the following method?

```php
public function onRaw(string $sql, bool $boolean = 'and')
```

And for convenience and consistency a `orOnRaw` method as well.

**Comment by Angel5a** [comment]
You can use `DB::raw()` in place of columns with the same result.

Personally if prefer an "ugly" `DB::raw()` over "neat" `onRaw()` in terms, that it make more obvious an unsafe structure of that code.
